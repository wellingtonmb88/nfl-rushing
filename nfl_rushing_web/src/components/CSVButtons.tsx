import React, { useState } from 'react';
import Button from '@material-ui/core/Button';
import ButtonGroup from '@material-ui/core/ButtonGroup';
import { useStyles } from '../styles'
import AppContext from '../context';

import { CSVDownload } from "react-csv";

export default function CSVButtons() {
  const classes = useStyles();
  const [downloadSelectedPlayers, setDownloadSelectedPlayers] = useState(false);
  const [downloadAllPlayers, setDownloadAllPlayers] = useState(false);
  const [downloadCurrentPlayers, setDownloadCurrentPlayers] = useState(false);

  return (
    <AppContext.Consumer>
      {({ players, selectedPlayers, allPlayers, getAllPlayers }) => (
        <div className={classes.buttonGroup}>
          <ButtonGroup variant="text" color="primary" aria-label="text primary button group">
            <Button disabled={selectedPlayers.length < 1} onClick={() => {
              setDownloadSelectedPlayers(true);
              setTimeout(() => {
                setDownloadSelectedPlayers(false);
              }, 200);
            }}>Export CSV (Selected Items)</Button>
            <Button disabled={players.length < 1} onClick={() => {
              setDownloadCurrentPlayers(true);
              setTimeout(() => {
                setDownloadCurrentPlayers(false);
              }, 200);
            }}>Export CSV (Current Page)</Button>
            <Button onClick={() => {
              getAllPlayers();
              setDownloadAllPlayers(true);
              setTimeout(() => {
                setDownloadAllPlayers(false);
              }, 500);
            }}>Export CSV (All Players Data)</Button>
          </ButtonGroup>

          {downloadCurrentPlayers ? <CSVDownload data={players} /> : null}
          {downloadSelectedPlayers ? <CSVDownload data={selectedPlayers} /> : null}
          {allPlayers.length > 0 && downloadAllPlayers ? <CSVDownload data={allPlayers} /> : null}
        </div>
      )
      }
    </AppContext.Consumer>
  );
}