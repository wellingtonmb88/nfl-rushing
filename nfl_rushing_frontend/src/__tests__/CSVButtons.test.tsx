import React from "react";
import { create } from "react-test-renderer";
import CSVButtons from '../components/CSVButtons'

describe("CSVButtons component", () => {
  test("CSVButtons with Snapshot", () => {
    const component = create(<CSVButtons/>);
    expect(component.toJSON()).toMatchSnapshot();
  });
});