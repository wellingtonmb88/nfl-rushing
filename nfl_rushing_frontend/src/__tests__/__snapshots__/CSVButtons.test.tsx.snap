// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CSVButtons component CSVButtons with Snapshot 1`] = `
<div
  className="makeStyles-buttonGroup-5"
>
  <div
    aria-label="text primary button group"
    className="MuiButtonGroup-root"
    role="group"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedText MuiButtonGroup-groupedTextHorizontal MuiButtonGroup-groupedTextPrimary MuiButton-textPrimary Mui-disabled Mui-disabled"
      disabled={true}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={-1}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Export CSV (Selected Items)
      </span>
    </button>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedText MuiButtonGroup-groupedTextHorizontal MuiButtonGroup-groupedTextPrimary MuiButton-textPrimary Mui-disabled Mui-disabled"
      disabled={true}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={-1}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Export CSV (Current Page)
      </span>
    </button>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedText MuiButtonGroup-groupedTextHorizontal MuiButtonGroup-groupedTextPrimary MuiButton-textPrimary"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Export CSV (All Players Data)
      </span>
    </button>
  </div>
</div>
`;
